solver/llbsolver/file/backend.go

```diff
@@ -266,7 +237,8 @@ func (fb *Backend) Mkdir(ctx context.Context, m, user, group fileoptypes.Mount,
 	}
 	defer lm.Unmount()
 
-	u, err := readUser(action.Owner, user, group)
+	// u, err := readUser(action.Owner, user, group, fb.Executor)
+	u, err := fb.readUserFn(action.Owner, user, group)
```

Comment:
This should not panic on `nil`.

Reply:
What do you think if we unexport `Backend{}` and have `NewFileOpBackend()` return an error if we pass in a nil callback. I think we should always pass in a callback even if it's a noop. This way we can avoid having to check if the read user function is nill whenever we want to use it, and simply check once in the "constructor".

Reply:
Just a nil check should do I think. Or defaulting to noop function.